src/Primitives/Search.purs

```diff
@@ -198,6 +219,19 @@ component =
       -- the parent to set the text.
       SearchReceiver i a -> H.modify (updateStore i.render id) *> pure a
 
+      CaptureFocus e a -> a <$ do
+        (Tuple _ st) <- getState
+        let el = either (const Nothing) Just <<< runExcept <<< readHTMLElement <<< toForeign <<< currentTarget <<< focusEventToEvent $ e
+        H.modify $ seeks _ { inputEl = el }
+        H.raise Focused
+
+      TriggerFocus a -> do
+        (Tuple _ st) <- getState
+        case st.inputEl of
+          Nothing -> pure a
+          Just el -> do
+            _ <- H.liftEff $ focus el
+            pure a
```

Comment:
We run into effectful cases like this often @crcornwell @whoadave @foresttoney. In these cases we can make things a bit cleaner with traverse:
```purescript
TriggerFocus a -> a <$ do
  (Tuple _ st) <- getState 
  traverse_ (H.liftEff <<< focus) st.inputEl
```